File: programming/java/SubsetAND.java
import java.util.Scanner;

class TestClass {
   public static void main(String args[] ) throws Exception
    {
                       Scanner input = new Scanner(System.in);
                       int T=input.nextInt();
     while(T-- >0){
       int z= input.nextInt();
       int n=input.nextInt();
       int a[]=new int[n];
       int flag=0;
       for(int i=0;i<n;i++){
            a[i]=input.nextInt();
       }
       for(int i=0;i<n;i++){
           if((z=(z&a[i]))==0){
                flag=1;
                break;
           }
       }
       if(flag==1){
               System.out.println("Yes");
       }
       else{
               System.out.println("No");
       }
    }
   }
}

File: programming/c/Subset_and.c
#include <stdio.h>

int main()
{
   int t,n;
   long int z,i;
   scanf("%d",&t);
        while(t--){
         int flag=0;
         scanf("%ld %d",&z,&n);
         long int ar[n];
               for(i=0;i<n;i++) {scanf("%ld",&ar[i]);
              // for(i=0;i<n;i++){
                                z=z&ar[i];
               	                if(z==0) flag=1;
               	                       //break;
               	                       // }
                               }
        if(flag == 1) printf("Yes\n");
        else printf("No\n");
        }
    return 0;
}

